### PR TITLE
refactor: rest APIs to use twenty orm

### LINKED ISSUE #3644: Make REST API use Twenty ORM direclty
**Objective**: Enhance the reliability and performance of the REST API.

## Technical Context
Currently, our REST API (documentation [here](https://twenty.com/developers/rest-api/core#)) is exposed via two dynamic endpoint sets:
- `rest/...`
- `rest/batch/...`
These endpoints are handled by two controllers: `rest-api-core.controller` and `rest-api-batch-core.controller`, both backed by `RestApiCoreService`.

The RestApiCoreService currently parses REST payloads, converts them into GraphQL payloads, and calls the GraphQL API—an inefficient process. This approach was initially necessary because querying workspaceSchemas in Twenty required pg_graphql.

However, we recently introduced TwentyORM (soon to be renamed WorkspaceORM), a lightweight layer on top of TypeORM, which can directly query workspaceSchemas.

Goal: Refactor the REST API to leverage TwentyORM directly.

Note: The GraphQL API is also being migrated to TwentyORM (replacing `pg_graphql`), so you can draw inspiration from the `GraphqlQueryRunnerService`.

## Key Goals:
Refactor the REST API to use TwentyORM directly.
Ensure the REST API is fully covered by integration tests.

## Technical inputs

We should take a lot of inspiration from what has been done in the new GraphqlQueryRunnerService. Advices:
1) split the code in services / utils to keep small and well scoped functional pieces
2) Overall logic: a) parse REST payload into TypeORM format, b) send TypeORM query, c) parse result back into REST response
3) All the pieces are already there: the current code is able to parse REST into graphql and Graphql into twentyORM, we "only" need to skip one step. It will likely mean to write everything from scratch but this should still speed a lot the process

### PR DESCRIPTION
**Pull request by DeepaPrasanna**
fixes #3644

This isn't the whole refactoring yet. I tried to refactor the `rest/batch/...` endpoint first. I added comments too for the part that might have to be included which were in `GraphqlQueryRunnerService`. Please let me know if I am moving in the right direction and If there is anything that I might have missed. 

Note: I will split the code in well scoped functionalities after knowing that I am moving in the right direction. Please excuse me until then.

### CODE REVIEW

**Comment by DeepaPrasanna** [comment]
Hello @charlesBochet , may I get your feedback? As this is an incomplete PR, I would love to work on that after the feedback. Thank you for reviewing :)

**Comment by FelixMalfait** [comment]
/award 1500 sorry we couldn't give more feedback earlier

**Comment by DeepaPrasanna** [comment]
> /award 1500 sorry we couldn't give more feedback earlier

No worries! Thank you :) I would like to continue this later after feedback

File: packages/twenty-server/src/engine/middlewares/rest-metadata.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      'SignUp',
+      'RenewToken',
+      'EmailPasswordResetLink',
+      'ValidatePasswordResetToken',
+      'UpdatePasswordViaResetToken',
+      'IntrospectionQuery',
+      'ExchangeAuthorizationCode',
+      'GetAuthorizationUrl',
+      'FindAvailableSSOIdentityProviders',
+    ];
```
Comment (martmull) [inline packages/twenty-server/src/engine/middlewares/rest-metadata.middleware.ts:43]: Please refactor this list with the list located in `packages/twenty-server/src/engine/middlewares/graphql-hydrate-request-from-token.middleware.ts`

File: packages/twenty-server/src/engine/middlewares/rest-metadata.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+      const dataSourcesMetadata =
+        await this.dataSourceService.getDataSourcesMetadataFromWorkspaceId(
+          data.workspace.id,
+        );
+
+      if (!dataSourcesMetadata || dataSourcesMetadata.length === 0) {
+        throw new Error('No data sources found');
+      }
+
```
Comment (martmull) [inline packages/twenty-server/src/engine/middlewares/rest-metadata.middleware.ts:76]: Lot of code in this new file is shared with `packages/twenty-server/src/engine/middlewares/graphql-hydrate-request-from-token.middleware.ts`. We should factorize this logic and create a base `HydrateRequestFromTokenMiddleware` middleware

File: packages/twenty-server/src/engine/middlewares/rest-metadata.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      req.user = data.user;
+      req.apiKey = data.apiKey;
+      req.workspace = data.workspace;
+      req.workspaceId = data.workspace.id;
+      req.workspaceMetadataVersion = metadataVersion;
+      req.workspaceMemberId = data.workspaceMemberId;
+    } catch (error) {
+      res.writeHead(200, { 'Content-Type': 'application/json' });
```
Comment (martmull) [inline packages/twenty-server/src/engine/middlewares/rest-metadata.middleware.ts:84]: for rest api, we don't want 200 error indeed

File: packages/twenty-server/src/engine/api/rest/core/query-builder/core-query-builder.factory.ts [hunk 4]
```diff
@@ -119,15 +134,69 @@ export class CoreQueryBuilderFactory {
...
+        id: In(objectRecords.generatedMaps.map((record) => record.id)),
+      })
+      .take(QUERY_MAX_RECORDS)
+      .getMany();
+
+    const upsertedRecords = formatResult(
+      nonFormattedUpsertedRecords,
+      objectMetadataMapItem,
+      objectMetadataMap,
+    );
```
Comment (martmull) [inline packages/twenty-server/src/engine/api/rest/core/query-builder/core-query-builder.factory.ts:196]: This code shared lot of code with `packages/twenty-server/src/engine/api/graphql/graphql-query-runner/resolvers/graphql-query-create-many-resolver.service.ts`
We must factorize that shared code

**Comment by martmull** [comment]
Hey @DeepaPrasanna, added some comments on your PR. I think this is the good way to go. I just see duplicated code that we must factorize

File: packages/twenty-server/src/engine/middlewares/rest-metadata.middleware.ts (hunk 3 above)
Comment (DeepaPrasanna) [reply packages/twenty-server/src/engine/middlewares/rest-metadata.middleware.ts:84]: yes indeed! I really missed that

**Comment by charlesBochet** [comment]
Closing this PR to keep history clear :)

**Pull request closed without merging**